### PR TITLE
Fix inconsistencies between attach_filter

### PR DESCRIPTION
**Pull request by gpotter2**
fixes https://github.com/secdev/scapy/issues/1655

### CODE REVIEW

File: scapy/arch/common.py [hunk 1]
```diff
@@ -10,15 +10,28 @@
...
 from ctypes import POINTER, Structure
 from ctypes import c_uint, c_uint32, c_ushort, c_ubyte
 from scapy.config import conf
+from scapy.data import MTU
+from scapy.error import log_loading, Scapy_Exception
 import scapy.modules.six as six
 
+# BOOT
+
+with os.popen("%s -V 2> /dev/null" % conf.prog.tcpdump) as _f:
```
Comment (guedou) [inline scapy/arch/common.py]: While you are modifying code, can you do the following ?
- rename _f to fdesc
- split the warning message to remove the noqa comment

File: scapy/arch/common.py [hunk 2]
```diff
@@ -95,3 +108,34 @@ def get_bpf_pointer(tcpdump_lines):
...
     # Create the BPF program
     return bpf_program(size, bip)
+
+
+def compile_filter(bpf_filter, iface):
+    """Asks Tcpdump to parse the filter, then build the matching
+    BPF bytecode using get_bpf_pointer.
+    """
+    if not TCPDUMP:
+        raise Scapy_Exception("Tcpdump is not available. Cannot use filter")
```
Comment (guedou) [inline scapy/arch/common.py]: s/Tcpdump/tcpdump/
s/filter/filter!/

File: scapy/arch/common.py [hunk 3]
```diff
@@ -95,3 +108,34 @@ def get_bpf_pointer(tcpdump_lines):
...
+    try:
+        p = subprocess.Popen([
```
Comment (guedou) [inline scapy/arch/common.py]: Could you use process instead of p ?

File: scapy/arch/common.py [hunk 4]
```diff
@@ -95,3 +108,34 @@ def get_bpf_pointer(tcpdump_lines):
...
+            "-p",
+            "-i", (conf.iface if iface is None else iface),
+            "-ddd",
+            "-s", str(MTU),
+            bpf_filter],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+    except OSError as ex:
+        raise Scapy_Exception(
```
Comment (guedou) [inline scapy/arch/common.py]: Can you make your exception message fit on a single line?

**Comment by gpotter2** [comment]
Done ! Should be good

File: scapy/arch/common.py [hunk 5]
```diff
@@ -10,15 +10,29 @@
...
 from ctypes import POINTER, Structure
 from ctypes import c_uint, c_uint32, c_ushort, c_ubyte
 from scapy.config import conf
+from scapy.data import MTU
+from scapy.error import log_loading, Scapy_Exception
 import scapy.modules.six as six
 
+# BOOT
+
+with os.popen("%s -V 2> /dev/null" % conf.prog.tcpdump) as fdesc:
```
Comment (p-l-) [inline scapy/arch/common.py]: Would you mind replacing that with a `subprocess.Popen()` call?

Comment (gpotter2) [reply scapy/arch/common.py]: Done !

**Comment by p-l-** [comment]
Three tiny `flake8` errors:
```
scapy/arch/common.py:22:1: F401 'scapy.error.log_loading' imported but unused
scapy/arch/common.py:27:1: E302 expected 2 blank lines, found 1
scapy/arch/common.py:33:1: E305 expected 2 blank lines after class or function definition, found 1
```

**Pull request merged**